### PR TITLE
Add TutorialCommand abstract class

### LINKED ISSUE #118: Update Mark, Unmark and Reset Commands
Refactor code to improve code quality

### PR DESCRIPTION
**Pull request by DKHB0207**
Added TutorialCommand Abstract Class which MarkCommand, UnmarkCommand and ResetCommand extends to remove unnecessary repeating of code. 
resolves #118

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ResetCommand.java [hunk 1]
```diff
@@ -51,98 +40,24 @@ public ResetCommand(Index index, Tutorial tutorial) {
...
+    protected String getUnnecessaryMessage() {
+        return Messages.MESSAGE_RESET_UNNECESSARY;
     }
 
     /**
      * Generates a command execution success message for the tutorials that were marked as not taken place.
      * {@code personToEdit}.
      */
-    private String generateSuccessMessage(Person personToEdit, String resetTutorials) {
+    protected String generateSuccessMessage(Person personToEdit, String resetTutorials) {
```
Comment (yockcheng) [inline src/main/java/seedu/address/logic/commands/ResetCommand.java:60]: Should we abstract this out too? Perhaps another method getSuccessMessage() that child classes override and we can put the functionality of generating success messages in the parent class instead.

**Review by yockcheng — APPROVED** [review]
LGTM overall and it really helps to cut down on unnecessary duplicated code. Just have a small suggestion below for your consideration.

File: src/main/java/seedu/address/logic/commands/ResetCommand.java (hunk 1 above)
Comment (DKHB0207) [reply src/main/java/seedu/address/logic/commands/ResetCommand.java]: Wow did not see that! Thanks for the suggestion, I have changed refactored the code to include this change. :)

**Review by yockcheng — APPROVED** [review]
Nice

**Pull request merged**